frontend/src/views/WheresMyGeneV2/components/HeatMap/index.tsx

```diff
@@ -208,30 +209,27 @@ export default memo(function HeatMap({
     return result;
   }, [selectedGeneExpressionSummariesByTissueName, geneNameToIndex]);
 
-  const initialDisplayedCellTypes = useMemo(
-    () =>
-      Object.entries(tissuesByName).reduce((acc, [_, { id }]) => {
-        if (
-          (filteredTissues.length > 0 && filteredTissues.includes(id)) ||
-          filteredTissues.length === 0
-        ) {
-          acc.add(id + id);
-        }
-        return acc;
-      }, new Set<string>()),
-    [tissuesByName, filteredTissues]
-  );
+  const initialDisplayedCellTypeIds = useMemo(() => {
+    return Object.values(tissuesByName).reduce((acc, { id }) => {
+      if (
+        (filteredTissueIds.length > 0 && filteredTissueIds.includes(id)) ||
+        filteredTissueIds.length === 0
+      ) {
+        acc.add(id + id);
+      }
+      return acc;
+    }, new Set<string>());
+  }, [tissuesByName, filteredTissueIds]);
 
   // set of tissue names that are visible and set of cell types that are visible
   // presence is visible
   const [displayedCellTypes, setDisplayedCellTypes] = useState<Set<string>>(
-    initialDisplayedCellTypes
+    initialDisplayedCellTypeIds
   );
 
   useEffect(() => {
-    setDisplayedCellTypes(initialDisplayedCellTypes);
-    setExpandedTissues(new Set<string>());
```

Comment:
no longer needs this, since we add `key={selectedOrganismId}` to the Heatmap, so it will automatically clear all states when the `key` changes